### PR TITLE
Remove mockRequestFromExample

### PR DESCRIPTION
**Pull request by rickschubert**
This pull request removes the `mockRequestFromExample` function. This function does not fit anymore with the current style of the repository given that we have decided that mock-inspect is an enabler of contract testing instead of implementing contract testing on its own.

[In my current blog article](https://rickschubert.net/blog/posts/mock-inspect-contract-testing/), I outline how `mockRequest` on its own can be used in conjunction with contracts to create contract testing completely without `mockRequestFromExample`.

This new versions should be published as a major change which is why I updated the package version as well.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "mock-inspect",
-    "version": "4.0.2",
+    "version": "5.0.0",
```
Comment (thomaschaplin) [inline package.json:3]: Please could you revert this change and instead run `npm version major` as this will update the changelog. In addition, it will show up as a separate commit to just bump the version.

**Review by thomaschaplin — CHANGES_REQUESTED** [review]
See comments, also do we not need the `Example` type anymore?

File: package.json (hunk 1 above)
Comment (rickschubert) [reply package.json:3]: I somehow missed this; I will do this later.

**Review by thomaschaplin — APPROVED** [review]
Thanks for doing this!
